File: Misc/GPS/serial_to_udp.rs
use std::env;

fn main() {

}

// mutable global variables
// serial port
let mut serial_port:str = "";
// baud rate
let mut baud_rate:u32 = 0;
// UDP port
let mut udp_port:u16 = 0;
// UDP host
let mut udp_host:str = "";

// Function that opens a UDP socket on a given port and host
fn open_udp_socket(int port, str host) {
    // create a UDP socket
    udp_socket = socket(AF_INET, SOCK_DGRAM, 0);
    // set the socket options
    setsockopt(udp_socket, SOL_SOCKET, SO_REUSEADDR, 1);
    // bind the socket to the given port and host
    let sockaddr_in addr = {
        sin_family: AF_INET,
        sin_port: htons(port),
        sin_addr: inet_addr(host),
        sin_zero: { 0, 0, 0, 0, 0, 0, 0, 0 }
    };
    bind(udp_socket, addr);
}

File: Misc/GPS/serial_to_udp.py
import serial
import socket
import argparse

if __name__ == '__main__':
    
    parser = argparse.ArgumentParser(description='Serial to UDP')
    parser.add_argument('-p', '--port', type=str, default='/dev/ttyUSB0',
                        help='serial port')
    parser.add_argument('-b', '--baud', type=int, default=115200,
                        help='baud rate')
    parser.add_argument('-u', '--udp', type=str, default=10101,
                        help='UDP port')

    args = parser.parse_args()

    ser = serial.Serial(args.port, args.baud)

    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)

    while True:
        data = []
        for i in range(9):
            data.append(ser.readline())
        data = b''.join(data)
        sock.sendto(data, ('localhost', int(args.udp)))